util/movement-types/src/lib.rs

```diff
@@ -22,34 +22,44 @@ impl Id {
 	}
 }
 
+impl AsRef<[u8]> for Id {
+	fn as_ref(&self) -> &[u8] {
+		&self.0
+	}
+}
+
 impl fmt::Display for Id {
 	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
 		write!(f, "{:?}", &self.0)
 	}
 }
 
 #[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct Transaction(pub Vec<u8>);
+pub struct Transaction {
+	pub data: Vec<u8>,
+	pub sequence_number: u64,
```

Comment:
Public struct members are not future-proof.
Consider making them private and providing a getter API; not necessarily in this PR though.

Reply:
I mean, it should be easy enough to add in this PR.